src/app/providers/search/search.ts

```diff
@@ -39,20 +39,11 @@ export class SearchProvider {
   }
 
   advancedSearch(advancedSearchParams: AdvancedSearchParams): Observable<SearchResultTestSchema[]> {
+    const searchParams = JSON.parse(JSON.stringify(advancedSearchParams));
```

Comment:
This will break advanced search, `dtcCode` will now not being sent and `costCode` will instead.

Also, we have now lost the ability to `encodeURIComponent`.